### PR TITLE
Improve Reactive Route guide

### PR DESCRIPTION
**Pull request by loicmathieu**
Some improvements to the reactive route guide.

I re-read the all guide recently and I find it a bit hard to read, don't know how we can improve it.

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 1]
```diff
@@ -771,7 +781,7 @@ Also see xref:openapi-swaggerui.adoc[the OpenAPI Guide].
 === Adding MicroProfile OpenAPI Annotations
 
 You can use link:https://github.com/eclipse/microprofile-open-api[MicroProfile OpenAPI] to better document your schema, 
-example, adding header info, or specifying the return type on `void` methods might be useful :
+by example, adding header info, or specifying the return type on `void` methods might be useful:
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:784]:
```suggestion
for instance, adding header info, or specifying the return type on `void` methods might be useful:
```
(or for example)

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 2]
```diff
@@ -149,7 +159,7 @@ String person() {
     // ...
 }
 ----
-<1> If the `accept` header matches `text/html` we set the content type automatically.
+<1> If the `accept` header matches `text/html` we set the content type automatically to `text/html`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:162]:
```suggestion
<1> If the `accept` header matches `text/html`, we set the content type automatically to `text/html`.
```

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 3]
```diff
@@ -140,7 +150,7 @@ public void route(RoutingContext rc) {
 ----
 <1> Each route can use different paths, methods...
 
-If no content-type header is set then we will try to use the most acceptable content type as defined by `io.vertx.ext.web.RoutingContext.getAcceptableContentType()`.
+If no content-type header is set,  then we will try to use the most acceptable content type by matching the accept header with the value of the `Route` produces attribute as defined by `io.vertx.ext.web.RoutingContext.getAcceptableContentType()`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:153]:
```suggestion
If no content-type header is set, then we will try to use the most acceptable content type by matching the accept header with the value of the `Route` produces attribute as defined by `io.vertx.ext.web.RoutingContext.getAcceptableContentType()`.
```

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 4]
```diff
@@ -91,14 +99,16 @@ public class MyDeclarativeRoutes {
 <3> The method gets a https://vertx.io/docs/apidocs/io/vertx/ext/web/RoutingContext.html[`RoutingContext`] as a parameter. From the `RoutingContext` you can retrieve the HTTP request (using `request()`) and write the response using `response().end(...)`.
 <4> If the annotated method does not return `void` the arguments are optional.
 <5> `RoutingExchange` is a convenient wrapper of `RoutingContext` which provides some useful methods.
+<6> The `RoutingExchange` is used to retrieve the request query parameter `name`.
+<7> The path define a parameter `name` which can be injected inside the method parameters using the annotation `@Param`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:103]:
```suggestion
<7> The path defines a parameter `name` which can be injected inside the method parameters using the annotation `@Param`.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks, I spotted a few things.

**Comment by loicmathieu** [comment]
Thanks @gsmet, team works ;), I applied your changes.

**Review by gsmet — APPROVED** [review]

**Pull request merged**